scripts/cmake/z_vcpkg_forward_output_variable.cmake

```
@@ -0,0 +1,34 @@
+#[===[.md:
+# z_vcpkg_forward_output_variable
+
+This macro is a simple forward-er to PARENT_SCOPE.
+
+```cmake
+z_vcpkg_forward_output_variable(output_var var_to_forward)
+```
+
+is equivalent to
+
+```cmake
+if(DEFINED output_var)
+    if(DEFINED value_var)
+        set("${output_var}" "${value_var}" PARENT_SCOPE)
+    else()
+        unset("${output_var}" PARENT_SCOPE)
+    endif()
+endif()
+```
+#]===]
```

````suggestion

Take special note that the first argument should be a local variable that has a value of the parent variable name. Most commonly, this local is the result of a pointer-out parameter to a function. If the variable in the first parameter is not defined, this function does nothing, simplifying functions with optional out parameters.

Most commonly, this should be used in cases like:
```cmake
function(my_function out_var)
    file(SHA512 "somefile.txt" local_var)
    z_vcpkg_forward_output_variable(out_var local_var)
endfunction()
```
#]===]
````